Sources/ClientRuntime/Networking/Http/HttpContext.swift

```diff
@@ -107,69 +101,77 @@ public class HttpContextBuilder {
         return self
     }
 
-    @discardableResult
-    public func withEncoder(value: RequestEncoder) -> HttpContextBuilder {
-        self.attributes.set(key: encoder, value: value)
-        return self
-    }
+//    @discardableResult
+//    public func withAuthScheme(value: AuthScheme) -> HttpContextBuilder {
+//        var authSchemes: Attributes
+//        if self.attributes.contains(key: AttributeKeys.authSchemes) {
+//            authSchemes = self.attributes.get(key: AttributeKeys.authSchemes)!
+//        } else {
+//            authSchemes = Attributes()
+//        }
+//        authSchemes.set(key: AttributeKey<AuthScheme>(name: "\(value.schemeId)"), value: value)
+//        self.attributes.set(key: AttributeKeys.authSchemes, value: authSchemes)
+//        return self
+//    }
 
     @discardableResult
-    public func withMethod(value: HttpMethodType) -> HttpContextBuilder {
-        self.attributes.set(key: method, value: value)
+    public func withDecoder(value: ResponseDecoder) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.decoder, value: value)
         return self
     }
 
     @discardableResult
-    public func withPath(value: String) -> HttpContextBuilder {
-        self.attributes.set(key: path, value: value)
+    public func withEncoder(value: RequestEncoder) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.encoder, value: value)
         return self
     }
 
     @discardableResult
     public func withHost(value: String) -> HttpContextBuilder {
-        self.attributes.set(key: host, value: value)
+        self.attributes.set(key: AttributeKeys.host, value: value)
         return self
     }
 
     @discardableResult
     public func withHostPrefix(value: String) -> HttpContextBuilder {
-        self.attributes.set(key: hostPrefix, value: value)
+        self.attributes.set(key: AttributeKeys.hostPrefix, value: value)
         return self
     }
 
     @discardableResult
-    public func withOperation(value: String) -> HttpContextBuilder {
-        self.attributes.set(key: operation, value: value)
-        return self
-    }
-
-    @discardableResult
-    public func withServiceName(value: String) -> HttpContextBuilder {
-        self.attributes.set(key: serviceName, value: value)
+    public func withIdempotencyTokenGenerator(value: IdempotencyTokenGenerator) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.idempotencyTokenGenerator, value: value)
         return self
     }
 
     @discardableResult
-    public func withDecoder(value: ResponseDecoder) -> HttpContextBuilder {
-        self.attributes.set(key: decoder, value: value)
+    public func withIdentityResolver(value: any IdentityResolver, type: IdentityType) -> HttpContextBuilder {
```

Comment:
What is the reason we are using `any` here again? I think we should prefer to use Generics but as they are provide type safety, type inference, and type casts for us but if we have a specific reason let's document that.

`withIdentityResolver<T: IdentityResolver>(value: T, type: IdentityType)`

Reply:
That's a good point - changed as suggested.